--- a/src/syncdb.php
+++ b/src/syncdb.php
@@ -176,6 +176,9 @@
         if (property_exists($endpoint, 'sql_log_bin') && !is_bool($endpoint->sql_log_bin)) {
             throw new \RuntimeException('invalid profile value in ' . $prefix . '.sql_log_bin');
         }
+        if (property_exists($endpoint, 'reset_definer') && !is_bool($endpoint->reset_definer)) {
+            throw new \RuntimeException('invalid profile value in ' . $prefix . '.reset_definer');
+        }
         if (!isset($endpoint->ssh) || $endpoint->ssh === false) {
             return;
         }
@@ -646,6 +649,15 @@
             echo ' (' . number_format(microtime(true) - $time_tmp, 2) . 's)';
             echo PHP_EOL;
 
+            if (($config->target->reset_definer ?? false) === true) {
+                self::executeCommand(
+                    'sed -E -i' . (self::getOs() === 'mac' ? " ''" : '') . ' -e ' . $sed_quote .
+                    's/^((CREATE( OR REPLACE)?( ALGORITHM=[A-Z]+)?)|(\/\*![0-9]+( CREATE\*\/ \/\*![0-9]+)?)) ' .
+                    'DEFINER=`([^`]|``)*`@`([^`]|``)*`/\1 DEFINER=CURRENT_USER/' . $sed_quote . ' ' . $tmp_filename,
+                    '--- RESETTING OBJECT DEFINERS...'
+                );
+            }
+
             // search / replace
             if (isset($config->replace)) {
                 $time_tmp = microtime(true);
